tests: drivers: pwm: frdm_mcxa153/mimxrt685_evk_cm33 : build failure

### Describe the bug

- frdm_mcxa153
- mimxrt685_evk_cm33

commit c30ad66e46a6e95e9ce6ba63c5f23d9848f758a5
Author: Felix Wang <fei.wang_3@nxp.com>
Date:   Mon Dec 15 09:33:11 2025 +0530

    tests: drivers: pwm: pwm_api: boards: Enable frdm_mcxa153 test
    
    Add device tree overlay to enable PWM API testing on the frdm_mcxa153
    board using ctimer0. The overlay configures:
    - PWM test alias pointing to ctimer0
    - Pin P2_12 (J2_12) as PWM output
    - ctimer0 in PWM mode with prescaler set to 1
    
    Signed-off-by: Felix Wang <fei.wang_3@nxp.com>

 .../pwm/pwm_api/boards/frdm_mcxa153.overlay        | 31 ++++++++++++++++++++++
 1 file changed, 31 insertions(+)
 create mode 100644 tests/drivers/pwm/pwm_api/boards/frdm_mcxa153.overlay

and

commit fa48bee6c478b0a0c42c9685177bffa5b1f9ad90
Author: Felix Wang <fei.wang_3@nxp.com>
Date:   Mon Dec 15 09:27:20 2025 +0530

    drivers: pwm: fix prescaler calculation for pwm_mcux_ctimer
    
    The prescaler divides the clock by (prescaler + 1), not by the prescaler
    value directly. Update the calculation to correctly account for this.
    
    Signed-off-by: Felix Wang <fei.wang_3@nxp.com>

 drivers/pwm/pwm_mcux_ctimer.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

### Regression

- [x] This is a regression.

### Steps to reproduce

```
west build -b frdm_mcxa153 tests/drivers/pwm
``` 
and
```
west build -b mimxrt685_evk/mimxrt685s/cm33 samples/basic/blinky_pwm
``` 


### Relevant log output

```shell
/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/__assert.h:124:25: note: in expansion of macro '__ASSERT_MSG_INFO' 124 
| __ASSERT_MSG_INFO(fmt, ##__VA_ARGS__); \ | ^~~~~~~~~~~~~~~~~/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/pwm/pwm_mcux_ctimer.c:211:9: note: in expansion of macro '__ASSERT' 211 
| __ASSERT((*cycles) > 0, "Invalid PWM frequency: cycles per second is 0(check clock rate | ^~~~~~~~/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/pwm/pwm_mcux_ctimer.c:212:31: warning: missing terminating " character 212 
| and prescaler)"); | ^/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/pwm/pwm_mcux_ctimer.c:212:31: error: missing terminating " character 212 
| and prescaler)"); 
| ^~~/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/pwm/pwm_mcux_ctimer.c:214:9
: error: expected ';' before 'return' 214 
| return err; 
| ^~~~~~/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/pwm/pwm_mcux_ctimer.c:215:1: warning: control reaches end of non-void function [-Wreturn-type] 215 
| } 
| ^ninja: build stopped: subcommand failed.
```

and for mimrt685_evk

```

mimxrt685_evk/mimxrt685s/cm33 
| [103/175]   Building C object   zephyr/drivers/gpio/CMakeFiles/drivers__gpio.dir/gpio_mcux_lpc.c.obj
[104/175]   Linking C static library zephyr/drivers/gpio/libdrivers__gpio.a
[105/175]   Building C object zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/common.c.obj
[106/175]   Building C object   zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/pinctrl_lpc_iocon.c.obj
[107/175]   Linking C static library   zephyr/drivers/pinctrl/libdrivers__pinctrl.a
[108/175] Building C object   zephyr/drivers/pwm/CMakeFiles/drivers__pwm.dir/pwm_mcux_ctimer.c.obj
FAILED:   zephyr/drivers/pwm/CMakeFiles/drivers__pwm.dir/pwm_mcux_ctimer.c.obj   hyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security   -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith   -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int   -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec   -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop   -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/samples/basic/blinky_pwm=CMAKE_SOURCE_DIR   -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr=ZEPHYR_BASE   -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_test_pipe=WEST_TOPDIR   -ffunction-sections -fdata-sections -mcmse -specs=picolibc.specs -std=c17 -MD   -MT zephyr/drivers/pwm/CMakeFiles/drivers__pwm.dir/pwm_mcux_ctimer.c.obj -MF   zephyr/drivers/pwm/CMakeFiles/drivers__pwm.dir/pwm_mcux_ctimer.c.obj.d -o   zephyr/drivers/pwm/CMakeFiles/drivers__pwm.dir/pwm_mcux_ctimer.c.obj -c   /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/pwm/pwm_mcux_ctimer.c/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/pwm/pwm_mcux_ctimer.c:   In function 'mcux_ctimer_pwm_get_cycles_per_sec':/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/pwm/pwm_mcux_ctimer.c:211:33:   warning: missing terminating " character 211 \
| __ASSERT((*cycles) >   0, "Invalid PWM frequency: cycles per second is 0(check clock rate \
|   ^/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/pwm/pwm_mcux_ctimer.c:212:31:   warning: missing terminating " character 212 \
| and prescaler)"); \
|   ^/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/pwm/pwm_mcux_ctimer.c:212:31:   error: missing terminating " character 212 \
| and prescaler)"); \
|   ^~~ninja: build stopped: subcommand failed.
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- OS: ubuntu
- toolchain: zephyr
- version: v4.3.0-3988-g3dfe5a98f6d9

### Additional Context

_No response_

--- a/drivers/pwm/pwm_mcux_ctimer.c
+++ b/drivers/pwm/pwm_mcux_ctimer.c
@@ -1,6 +1,6 @@
 /*
  * (c) Meta Platforms, Inc. and affiliates.
- * Copyright 2025 NXP
+ * Copyright 2025-2026 NXP
  *
  * SPDX-License-Identifier: Apache-2.0
  */
@@ -208,8 +208,8 @@ static int mcux_ctimer_pwm_get_cycles_per_sec(const struct device *dev, uint32_t
 	}
 
 	*cycles /= (uint64_t)config->prescale + 1;
-	__ASSERT((*cycles) > 0, "Invalid PWM frequency: cycles per second is 0(check clock rate
-		and prescaler)");
+	__ASSERT((*cycles) > 0, "Invalid PWM frequency: cycles per second is 0(check clock rate "
+				"and prescaler)");
 
 	return err;
 }

